### ISSUE TITLE
Message C0330 reported for PEP-8 compliant 8-space hanging indent

### ISSUE THREAD

**Issue by bignose-debian** [PyCQA/pylint#741]
The following code examples use 8-space hanging indentation:

```python

# More indentation included to distinguish this from the rest.
def long_function_name(
        var_one, var_two, var_three,
        var_four):
    print(var_one)

def short_func_name(foo):
    # Add some extra indentation on the conditional continuation line.
    if (this_is_one_thing
            and that_is_another_thing):
        do_something()

# Some normal multi-line statements with double-indented continuation lines.
large_collection = [
        "spam",
        "eggs",
        "beans",
        ]
if True:
    pass # Normal statement-block indentation.

long_function_name(
        "1", "2", "3", "4")
if True:
    pass # Normal statement-block indentation.

concatenated_text = (
        "spam"
        "eggs"
        "beans")
if True:
    pass # Normal statement-block indentation.
```

These are all PEP 8 conformant. The relevant [PEP 8 section on code indentation](http://legacy.python.org/dev/peps/pep-0008/#indentation) states:

> further indentation should be used to clearly distinguish itself as a continuation line.

But Pylint 1.4.4 fails them:

```
$ pylint --version
No config file found, using default configuration
pylint 1.4.4,
astroid 1.3.8, common 1.0.2
Python 2.7.11 (default, Dec  9 2015, 00:29:25)
[GCC 5.3.1 20151205]

$ pylint --disable=W --disable=E --disable=C --enable=C0330 ./pylint_issue_153.py
No config file found, using default configuration
************* Module pylint_issue_153
C: 15, 0: Wrong hanging indentation.
        "spam",
    |   ^ (bad-continuation)
C: 16, 0: Wrong hanging indentation.
        "eggs",
    |   ^ (bad-continuation)
C: 17, 0: Wrong hanging indentation.
        "beans",
    |   ^ (bad-continuation)
C: 18, 0: Wrong hanging indentation.
        ]
|   |   ^ (bad-continuation)
C: 23, 0: Wrong hanging indentation.
        "1", "2", "3", "4")
    |   ^ (bad-continuation)
C: 28, 0: Wrong hanging indentation.
        "spam"
    |   ^ (bad-continuation)
```

The indentation checker should, by default, allow all PEP 8 conformant hanging indentation.

**Comment by bignose-debian** [comment]
This was reported earlier at issue#153, with test cases showing it is not fixed. But was closed, so I'm reporting it again with full information.

**Comment by The-Compiler** [comment]
I don't think your example is pep8 compliant - from PEP8:

```python
# More indentation included to distinguish this from the rest.
def long_function_name(
        var_one, var_two, var_three,
        var_four):
    print(var_one)

# Hanging indents should add a level.
foo = long_function_name(
    var_one, var_two,
    var_three, var_four)
```

And pylint enforces those two examples just right - for `long_function_name` and `short_func_name` it didn't complain (as there *was* ambiguity which required indenting one more level), and for the others it did, because there was no ambiguity, so you shouldn't indent another level.

**Comment by vencabot** [comment]
bignose-debian's example is pep8 compliant, ala:

"The 4-space rule is optional for continuation lines.

Optional:

```
# Hanging indents *may* be indented to other than 4 spaces.
foo = long_function_name(
  var_one, var_two,
  var_three, var_four)
```
"

I, like bignose-debian, prefer that my first hanging indent go 8 spaces deep. This makes it clearly distinguishable, at a glance, from a logical code-block (even when a code-block isn't DIRECTLY neighboring, where the need is even more obvious).

This is also the default auto-indenting behavior when editing Python files in vim; the first hanging indent is 8 spaces deep (immediately distinguishable from a codeblock), and other nested indents deepen by 4 spaces each time, which I would say is evident of its being a common preference.

PEP8 recommends 'more indentation' for the sake of distinguishing hanging indents from code-blocks, and then it further goes on to say that hanging indents may be indented to other than 4 spaces. I don't see why pylint should yell at me for indenting to a clean 8 spaces.

**Comment by slinkp** [comment]
I'll just note that flake8 complains if you *don't* indent continuation lines extra, while pylint complains if you *do*.  I believe flake8 interprets pep8 correctly here.

```py
def a_function(*args, **kw):
    pass


def another_function(*args, **kw):
    pass


def pylint_happy_flake8_sad():
    """
    pylint likes this.
    But flake8 says:
    '...:18:9: E129 visually indented line with same indent as next logical
    line'
    """
    with a_function(
        "a longish string that would blow out line length adisufopsad",
        "another positional param",
        named_param="yep"):
        another_function()


def pylint_sad_flake8_happy():
    """
    flake8 likes this.
    But pylint says:
    C: 30, 0: Wrong hanging indentation (remove 4 spaces).
            "a longish string that would blow out line length adisufopsad",
        |   ^ (bad-continuation)
    """
    with a_function(
            "a longish string that would blow out line length adisufopsad",
            "another positional param",
            named_param="yep"):
        another_function()
```
